sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -24,6 +24,9 @@ namespace Azure.Core.Pipeline
         private readonly ActivityAdapter? _activityAdapter;
         private readonly bool _suppressNestedClientActivities;
 
+#if NET6_0_OR_GREATER
```

Comment:
Can we remove these `#if`s like we do elsewhere?

Reply:
Not including this breaks the build in a number of projects with errors as if the `trimmingattributes.cs` file isn't there.

I ran into a similar issue when I was updating the dependency on `System.Diagnostics.DiagnosticSource`, but basically the way our repo is structured, libraries use package references on Azure.Core, but then pull in files individually when using shared source. Since this is part of the shared source, I _think_ we have to wait until Azure.Core releases with trimmingattributes.cs. I had to do the same thing when updating DiagnosticSource (release with the package upgrade, then make the changes in the file, then release again).